resdk/tests/functional/duplication/e2e_duplication.py

```diff
@@ -0,0 +1,39 @@
+# pylint: disable=missing-docstring
+from resdk.exceptions import ResolweServerError
+from resdk.tests.functional.base import BaseResdkFunctionalTest
+
+
+class TestDuplication(BaseResdkFunctionalTest):
+
+    def test_collection_duplication(self):
+        collection = self.res.collection.create(name='Test collection')
+        duplicate = collection.duplicate()
+        self.assertEqual(duplicate.name, 'Copy of Test collection')
+
+    def test_sample_duplication(self):
+        sample = self.res.sample.create(name='Test sample')
+        duplicate = sample.duplicate()
+        self.assertEqual(duplicate.name, 'Copy of Test sample')
+
+    def test_sample_duplication_perms(self):
+        collection = self.user_res.collection.create()
+        sample = self.user_res.sample.create(name='Test sample')
+        collection.add_samples(sample)
+
+        collection.permissions.remove_user(self.user_res.auth.username, 'add')
+        # TODO is it ok that no exception is raised when duplicating a sample
+        # with 'inherit_collection' set to True while the user has no 'add' permission
+        # on the collection?
+        # What happens is that the sample is duplicated just fine, but is also NOT
+        # added to the collection (without an exception being raised).
+        duplicate = sample.duplicate(inherit_collection=True)
+        self.assertEqual(len(collection.samples), 1)
+        self.assertEqual(collection.samples[0].name, 'Test sample')
+        self.assertEqual(duplicate.name, f'Copy of {sample.name}')
+
+    def test_data_duplication(self):
+        data = self.res.run(slug='test-sleep-progress', input={'t': 1})
+        # Let's not wait for processing to finish and
+        # check the expected exception to be raised.
+        with self.assertRaisesRegex(ResolweServerError, 'done or error status to be duplicated'):
```

Comment:
Why do you expect error during duplication?

Reply:
Read the comment above :)

Reply:
Duplication is allowed only for data objects that have `OK` or `ER` status. In other cases exception is raised and since we only care if endpoint works as expected, we check that.

Reply:
OK, makes sense.

Reply:
Maybe add an explicit comment that error is expected because data will not have an OK/ER status. I am familiar with Resolwe duplication mechanism, but still haven't made the connection.

Reply:
Added an extra comment below.